src/TurbulenceClosures/turbulence_closure_implementations/CATKEVerticalDiffusivities/time_step_turbulent_kinetic_energy.jl

```diff
@@ -0,0 +1,206 @@
+using Oceananigans: fields
+using Oceananigans.Advection: div_Uc, U_dot_∇u, U_dot_∇v
+using Oceananigans.Fields: immersed_boundary_condition
+using Oceananigans.Grids: active_interior_map
+using Oceananigans.BoundaryConditions: apply_x_bcs!, apply_y_bcs!, apply_z_bcs!
+using Oceananigans.TimeSteppers: store_field_tendencies!, ab2_step_field!, implicit_step!
+using Oceananigans.TurbulenceClosures: ∇_dot_qᶜ, immersed_∇_dot_qᶜ, hydrostatic_turbulent_kinetic_energy_tendency
+using CUDA
+
+tke_time_step(closure::CATKEVerticalDiffusivity) = closure.turbulent_kinetic_energy_time_step
+
+function tke_time_step(closure_array::AbstractArray)
+    # assume they are all the same
+    closure = CUDA.@allowscalar closure_array[1, 1]
+    return tke_time_step(closure)
+end
+
+function time_step_turbulent_kinetic_energy!(model)
+
+    # TODO: properly handle closure tuples
+    closure = model.closure
+
+    e = model.tracers.e
+    arch = model.architecture
+    grid = model.grid
+    Gⁿe = model.timestepper.Gⁿ.e
+    G⁻e = model.timestepper.G⁻.e
+
+    diffusivity_fields = model.diffusivity_fields
+    κe = diffusivity_fields.κe
+    Le = diffusivity_fields.Le
+    previous_velocities = diffusivity_fields.previous_velocities
+    tracer_index = findfirst(k -> k == :e, keys(model.tracers))
+    implicit_solver = model.timestepper.implicit_solver
+
+    Δt = model.clock.last_Δt
+    Δτ = tke_time_step(closure)
```

Comment:
is Δτ the tilmestep for `e`? and is it smaller than Δt?
is this is the first place it appears perhaps a comment?

Reply:
> is Δτ the tilmestep for e?

yes. But isn't `Δτ = tke_time_step(closure)` self explanatory? (tke is e)

> and is it smaller than Δt?

It is either `nothing`, or some number... when taking the tke time-step, it is limited by the large-scale time-step. So it is either `nothing` (in which case we take the same value as `Δt`), or it is either equal to or smaller than `Δt` if explicitly prescribed.